tests/python/unittest/test_sparse_operator.py

```diff
@@ -175,6 +179,32 @@ def check_csr_slice(shape, slice_input):
     check_csr_slice(shape, True)
     check_csr_slice(shape, False)
 
+
+def test_sparse_retain():
+    for num_rows in range(1, 40):
+        num_cols = 3
+        shape = (num_rows, num_cols)
+        rsp, _ = rand_sparse_ndarray(shape=shape, storage_type='row_sparse', density=0.5)
+        length = np.random.randint(1, num_rows + 1)
+        import random
```

Comment:
We seldom import `random`. You can use rnd.randint() instead, which is from numpy. There's also sampling logic in `def rand_sparse_ndarray`